great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
+                )
+            action_list.append(
+                {
+                    "name": "open_lineage",
+                    "action": {
+                        "class_name": "OpenLineageValidationAction",
+                        "module_name": "openlineage.common.provider.great_expectations",
+                        "openlineage_host": openlineage_host,
+                        "openlineage_apiKey": openlineage_api_key,
+                        "openlineage_namespace": openlineage_namespace,
+                        "job_name": f"validate_{self.task_id}",
+                    },
+                }
+            )
+        return action_list
+
+    def build_default_checkpoint_config(self):
+        """Builds a default checkpoint with default values."""
+        self.run_name = (
+            self.run_name
+            if self.run_name
+            else f"{self.task_id}_{datetime.now().strftime('%Y-%m-%d::%H:%M:%S')}"
+        )
+        checkpoint_config = CheckpointConfig(
+            name=self.checkpoint_name,
+            config_version=1.0,
+            template_name=None,
+            module_name="great_expectations.checkpoint",
+            class_name="Checkpoint",
+            run_name_template=self.run_name,
+            expectation_suite_name=self.expectation_suite_name,
+            batch_request=None,
+            action_list=self.build_default_action_list(),
+            evaluation_parameters={},
+            runtime_configuration={},
+            validations=None,
+            profilers=[],
+            ge_cloud_id=None,
+            expectation_suite_ge_cloud_id=None,
+        ).to_json_dict()
+        filtered_config = {k: v for k, v in checkpoint_config.items() if v}
+
+        return filtered_config
+
+    def execute(self, context: "Context") -> Union[CheckpointResult, Dict[str, Any]]:
+        """
+        Determines whether a checkpoint exists or need to be built, then
+        runs the resulting checkpoint.
+        """
         self.log.info("Running validation with Great Expectations...")
+        self.conn = BaseHook.get_connection(self.conn_id) if self.conn_id else None
+        self.conn_type = self.conn.conn_type if self.conn else None
 
-        # Instantiate the Data Context
-        self.log.info("Ensuring data context is valid...")
+        self.log.info("Instantiating Data Context...")
+        if self.data_asset_name:
+            self.build_runtime_datasources()
         if self.data_context_root_dir:
-            self.data_context: BaseDataContext = ge.data_context.DataContext(
+            self.data_context = ge.data_context.DataContext(
                 context_root_dir=self.data_context_root_dir
             )
         else:
-            self.data_context: BaseDataContext = BaseDataContext(
-                project_config=self.data_context_config
-            )
+            self.data_context = BaseDataContext(project_config=self.data_context_config)
```

Comment:
❤️

Reply:
If we do, then we'd also have to write the datasource to the file if someone passes in `data_context_root_dir`, and we just want this to be a temporary datasource.

Reply:
@denimalpaca Got it.  Makes sense.  It would be nice to put in a comment or a docstring explaining this rationale. 
 Thanks.